cpg_pipes/providers/cpg/status.py

```diff
@@ -0,0 +1,127 @@
+"""
+CPG implementation of status reporter.
+"""
+
+from textwrap import dedent
+
+from hailtop.batch.job import Job
+from hailtop.batch import Batch
+
+from ... import images, Path
+from ...hb.command import wrap_command
+from ...targets import Target
+from ..status import (
+    AnalysisStatus,
+    StatusReporterError,
+    StatusReporter,
+)
+from .smdb import SMDB, SmdbError
+
+
+class SmdbStatusReporter(StatusReporter):
+    """
+    Job status reporter. Works through creating and updating sample-metadata
+    database Analysis entries.
+    """
+
+    def __init__(self, smdb: SMDB):
+        self.smdb = smdb
+
+    def add_updaters_jobs(
+        self,
+        b: Batch,
+        output: Path | dict[str, Path],
+        analysis_type: str,
+        target: Target,
+        jobs: list[Job] | None = None,
+    ) -> list[Job]:
+        """
+        Create "queued" analysis and insert "in_progress", and "completed" updater jobs.
+        """
+        if isinstance(output, dict):
+            raise StatusReporterError(
+                'SmdbStatusReporter only supports a single Path ' 'as an output data.'
+            )
+        if not jobs:
+            return []
+        # Interacting with the sample metadata server:
+        # 1. Create a "queued" analysis
+        aid = self.smdb.create_analysis(
+            output=str(output),
+            type_=analysis_type,
+            status='queued',
+            sample_ids=[s.id for s in target.get_samples()],
+        )
+        if aid is None:
+            raise StatusReporterError(
+                'SmdbStatusReporter error: failed to create analysis'
+            )
+        # 2. Queue a job that updates the status to "in-progress"
+        in_progress_j = self.add_status_updater_job(
+            b,
+            analysis_id=aid,
+            status=AnalysisStatus.IN_PROGRESS,
+            analysis_type=analysis_type,
+            job_attrs=target.get_job_attrs(),
+        )
+        # 2. Queue a job that updates the status to "completed"
+        completed_j = self.add_status_updater_job(
+            b,
+            analysis_id=aid,
+            status=AnalysisStatus.COMPLETED,
+            analysis_type=analysis_type,
+            job_attrs=target.get_job_attrs(),
+        )
+
+        in_progress_j.depends_on(jobs[0])
+        completed_j.depends_on(*jobs)
+        return [in_progress_j] + jobs + [completed_j]
```

Comment:
This syntax is fine, but also want to point out that this exists because I like the destructuring more

```suggestion
        return [in_progress_j, *jobs, completed_j]
```

Reply:
Yeah, it looks better